geomesa-gt/geomesa-gt-partitioning/src/test/scala/org/locationtech/geomesa/gt/partition/postgis/PartitionedPostgisDataStoreTest.scala

```diff
@@ -0,0 +1,158 @@
+
+import com.typesafe.scalalogging.LazyLogging
+import org.geotools.data.postgis.PostGISPSDialect
+import org.geotools.data.{DataStoreFinder, DefaultTransaction, Transaction}
+import org.geotools.filter.identity.FeatureIdImpl
+import org.geotools.filter.text.ecql.ECQL
+import org.geotools.jdbc.JDBCDataStore
+import org.geotools.util.factory.Hints
+import org.junit.runner.RunWith
+import org.locationtech.geomesa.gt.partition.postgis.dialect.PartitionedPostgisDialect
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.locationtech.geomesa.utils.io.WithClose
+import org.locationtech.geomesa.utils.text.WKTUtils
+import org.specs2.mutable.Specification
+import org.specs2.runner.JUnitRunner
+
+import scala.util.control.NonFatal
+
+@RunWith(classOf[JUnitRunner])
+class PartitionedPostgisDataStoreTest extends Specification with LazyLogging {
+
+  import scala.collection.JavaConverters._
+
+  val hours = 1
+  val spec =
+    "name:String,age:Int,dtg:Date,*geom:Point:srid=4326;" +
+        Seq(
+          s"pg.partitions.interval.hours=$hours",
+          "pg.partitions.cron.minute=0"/*,
+          "pg.partitions.max=2",
+          "pg.partitions.tablespace.wa=wa",
+          "pg.partitions.tablespace.wa-partitions=wapa",
+          "pg.partitions.tablespace.main=main"*/
+        ).mkString(",")
+
+  val methods =
+    Methods(
+      create = true,
+      write = true,
+      update = false,
+      delete = false,
+      remove = false
+    )
+
+  lazy val sft = SimpleFeatureTypes.createType(s"test", spec)
+
+  "PartitionedPostgisDataStore" should {
+    "work" in {
+      skipped("requires postgis instance")
+      val params =
+        Map(
+          "dbtype"   -> PartitionedPostgisDataStoreParams.DbType.sample,
+          "host"     -> "localhost",
+          "port"     -> "5432",
+          "schema"   -> "public",
+          "database" -> "geodata",
+          "user"     -> "postgres",
+          "passwd"   -> "gimmee",
+          "Batch insert size"  -> "10",
+          "Commit size"        -> "20",
+          "preparedStatements" -> "true"
+        )
+
+      val ds = DataStoreFinder.getDataStore(params.asJava)
+
+      try {
+        ds must not(beNull)
+        ds must beAnInstanceOf[JDBCDataStore]
+
+        logger.info(s"Existing type names: ${ds.getTypeNames.mkString(", ")}")
+
+        if (methods.create) {
+          ds.createSchema(sft)
+        } else {
+          WithClose(ds.asInstanceOf[JDBCDataStore].getConnection(Transaction.AUTO_COMMIT)) { cx =>
+            val dialect = ds.asInstanceOf[JDBCDataStore].dialect match {
+              case p: PartitionedPostgisDialect => p
+              case p: PostGISPSDialect =>
+                val m = p.getClass.getDeclaredMethod("getDelegate")
+                m.setAccessible(true)
+                m.invoke(p).asInstanceOf[PartitionedPostgisDialect]
+            }
+            dialect.recreateFunctions("public", sft, cx)
```

Comment:
*blinks* Ok, I might need more coffee or an explanation for the use of reflection here.  What's the biggest goal of this block?

Reply:
the JDBC data store is very locked down. it's a `final` class, and when using prepared statements you can't even access the dialect directly. this was just a bit of hackery to expose the custom dialect, so that I can re-create the plpgsql procedures without dropping and re-creating the schema.

Reply:
Yeah, true, the JDBC DS is locked down.